client/repo.c

```diff
@@ -633,28 +641,6 @@ TDNFDownloadUsingMetalinkResources(
     goto cleanup;
```

Comment:
in case of error we should see if pszRepoMDUrl is allocated, we should free.

Reply:
Ok, I see pszRepoMDUrl is getting freed by the caller, but in case there is an error then we should free pszRepoMDUrl and set ppszRepoMDUrl to NULL.